pkg/ddevapp/compose_yaml.go

```diff
@@ -132,6 +133,37 @@ func (app *DdevApp) ReadDockerComposeYAML() error {
 	return nil
 }
 
+// XDdevExtension represents the x-ddev extension data in docker-compose files
+type XDdevExtension struct {
+	DescribeURLPort string `mapstructure:"describe-url-port"`
+	DescribeInfo    string `mapstructure:"describe-info"`
+	Shell           string `mapstructure:"shell"`
+}
+
+// GetXDdevExtension retrieves the x-ddev extension for a given service from the ComposeYaml
+func (app *DdevApp) GetXDdevExtension(serviceName string) XDdevExtension {
+	var xDdev XDdevExtension
+	if app.ComposeYaml != nil && app.ComposeYaml.Services != nil {
+		if composeService, ok := app.ComposeYaml.Services[serviceName]; ok {
+			if found, err := composeService.Extensions.Get("x-ddev", &xDdev); err == nil && found {
+				// Trim whitespace from all string fields
+				xDdev.DescribeInfo = strings.TrimSpace(xDdev.DescribeInfo)
+				xDdev.DescribeURLPort = strings.TrimSpace(xDdev.DescribeURLPort)
+				xDdev.Shell = strings.TrimSpace(xDdev.Shell)
+			}
+		}
+	}
+	// Always default to bash for web and db services
+	if serviceName == "web" || serviceName == "db" {
```

Comment:
What if they want to use zsh? Can't override? Maybe these stanzas should go before the search for overrides.